pkg/collector/osmlogs_collector.go

```diff
@@ -73,70 +59,49 @@ func (collector *OsmLogsCollector) Collect() error {
 		if err != nil {
 			return err
 		}
-		collectDataFromNamespaces(collector, namespacesInMesh, rootPath, meshName)
-		collectControllerLogs(collector, rootPath, meshName)
+		if err = collectDataFromNamespaces(collector, namespacesInMesh, rootPath, meshName); err != nil {
+			return err
```

Comment:
This should be logged instead of exiting early. Failure to collect data from namespaces shouldn't stop collecting of controller logs.